### PR TITLE
Unlimited custom domain aliases

### LINKED ISSUE #5584: Unlimited custom domain mail addresses
As a user who owns a custom domain, I want to be able to define as many email addresses as I want with that domain, so that I can take full advantage of my domain name.

## AC
* Custom domain mail addresses are not counted as used email alias for plans with unlimited custom domain alias support.
* Behavior for Tutanota domain aliases does not change
* Unlimited custom domain alias feature is available for all new paid plans
* Numbers of used / available aliases are shown in the settings including a hint that custom domain aliases do not count towards the email address limit.

## Additional info
* Showing this feature in the plan feature list (website and client) will be handled in a separate issue.
* Server branch is unlimited-custom-addresses-5584

## Tasks
- [x] Adapt settings to show the number of used Tutanota email addresses and info about what affects limit
- [x] Add new translations to Phrase
- [x] Adapt server side to allow this
- [x] Make sure downgrading checks for all kinds of aliases

## Test Notes
* [ ] it's not possible to downgrade to free if there are any extra addresses enabled
* [ ] the number of used / enabled addresses does not change for new plans when adding or deleting custom domain addresses
* [ ] the number does change as expected when adding / disabling / enabling tutanota addresses for new and legacy plans
* [ ] it's possible to add more custom domain addresses if the limit is reached for tutanota addresses on new plans
* [ ] it's still possible to reassign any addresses to a different user when the limit is reached on the sending user
* [ ] it's still possible to reassign any addresses to a different user when the limit is reached for the customer on legacy plans
* [ ] it's only possible to reassign an address to a different user when the user's "used" limit is not reached (new plans)
* [ ] for legacy plan customers, the custom domain addresses still count towards the limit
* [ ] it is not possible to switch to a plan with less aliases if the "used" aliases count is higher (new plans and legacy)

### PR DESCRIPTION
**Pull request by charlag**
close #5584

tutadb  branch unlimited-custom-addresses-5584 (!463) 
website branch unlimited-custom-domain-aliases-5584 (!103)

### CODE REVIEW

**Comment by charlag** [comment]
aside from pulling the translations once they are ready it is done. Downgrades should be checked more thoroughly perhaps.

File: src/translations/de.ts [hunk 1]
```diff
@@ -1554,6 +1554,7 @@ export default {
 		"yourCalendars_label": "Deine Kalender",
 		"yourFolders_action": "DEINE ORDNER",
 		"yourMessage_label": "Deine Nachricht",
-		"you_label": "Du"
+		"you_label": "Du",
+		"mailAddressInfo_msg": "Eigene Domänenadressen werden nicht auf das Limit angerechnet. Deaktivierte E-Mail-Adressen können einem anderen Benutzer oder einer Mailbox zugewiesen werden."
```
Comment (ganthern) [inline src/translations/de.ts:1558]: I'd rather say

> Adressen eigener Domains werden nicht auf das Limit angerechnet. Deaktivierte E-Mail-Adressen können einem anderen Benutzer oder einer Mailbox zugewiesen werden.

File: src/settings/mailaddress/MailAddressTableModel.ts [hunk 2]
```diff
@@ -42,13 +42,17 @@ export interface MailAddressNameChanger {
...
 export class MailAddressTableModel {
-	readonly redraw = stream<void>()
 	private nameMappings: AddressToName | null = null
+	aliasCount: MailAddressAliasServiceReturn | null = null
 
 	init: () => Promise<void> = lazyMemoized(async () => {
 		this.eventController.addEntityListener(this.entityEventsReceived)
 		await this.loadNames()
 		this.redraw()
+		// FIXME do we want to call this service every time users open settings?
```
Comment (ganthern) [inline src/settings/mailaddress/MailAddressTableModel.ts:52]: fixme

**Comment by ganthern** [comment]
still needs test notes

**Comment by KernelBypass** [comment]
Hi,

Do you guys have an ETA for this feature release?

Thanks!

**Comment by vitoreiji** [comment]
> Do you guys have an ETA for this feature release?

We can't give out ETAs because it's ready when it's ready, but I can tell you we are working on it.

**Comment by KernelBypass** [comment]
> > Do you guys have an ETA for this feature release?
> 
> We can't give out ETAs because it's ready when it's ready, but I can tell you we are working on it.

Which plans are going to benefit from this feature? 
Thanks

**Comment by vitoreiji** [comment]
> Which plans are going to benefit from this feature? Thanks

All the new ones: Revolutionary, Legend, Essential, Advanced and Unlimited.

**Comment by KernelBypass** [comment]
> > Which plans are going to benefit from this feature? Thanks
> 
> All the new ones: Revolutionary, Legend, Essential, Advanced and Unlimited.

so there will be no limit on number of aliases when using custom domains regardless of how expensive the plan is (apart from free plan) -- only the number of domains, correct?

**Comment by vitoreiji** [comment]
> so there will be no limit on number of aliases when using custom domains regardless of how expensive the plan is (apart from free plan) -- only the number of domains, correct?
> 
> We'll be able to have even a 1000 aliases on a custom domain on any paid plan?

That is what is planned for now, but I cannot 100% guarantee that this won't change until release time.

File: src/settings/mailaddress/MailAddressTable.ts [hunk 3]
```diff
@@ -9,24 +9,25 @@ import { Icons } from "../../gui/base/icons/Icons.js"
...
 import { showAddAliasDialog } from "./AddAliasDialog.js"
 import { locator } from "../../api/main/MainLocator.js"
+import { UpgradeRequiredError } from "../../api/main/UpgradeRequiredError.js"
+import Stream from "mithril/stream"
 
 assertMainOrNode()
 
 export type MailAddressTableAttrs = {
 	model: MailAddressTableModel
+	expanded: Stream<boolean>
```
Comment (charlag) [inline src/settings/mailaddress/MailAddressTable.ts:26]: I consider streams in attrs antipattern

Comment (charlag) [reply src/settings/mailaddress/MailAddressTable.ts:26]: I will fix it

Comment (ganthern) [reply src/settings/mailaddress/MailAddressTable.ts]: ... how did that get there then

Comment (charlag) [reply src/settings/mailaddress/MailAddressTable.ts]: huh? it was @armhub and @nokhub

Comment (ganthern) [reply src/settings/mailaddress/MailAddressTable.ts]: github is showing this before the commits that added this code

Comment (ganthern) [reply src/settings/mailaddress/MailAddressTable.ts]: I'm not saying I don't believe you, I'm saying github is messing up

Comment (charlag) [reply src/settings/mailaddress/MailAddressTable.ts]: I was just trying to figure out where it comes from

**Pull request merged**